### PR TITLE
Fix version going blank after logging in

### PR DESCRIPTION
**Pull request by dbkr**
Don't use replaceState in MatrixClient: there's lots of stuff in
MatrixClient's state now (including the app version) so replacing
the entire state doesn't really make sense (and also blows away
all of the nice defaults we set in getInitialState). Instead,
setState of the things we actually care about wherever we used
replaceState.

Also add a couple of state variables to getInitialState that were
missing.

Fixes https://github.com/vector-im/vector-web/issues/2322

### CODE REVIEW

File: src/components/structures/MatrixChat.js [hunk 1]
```diff
@@ -105,6 +105,9 @@ module.exports = React.createClass({
 
             version: null,
             newVersion: null,
+
+            upgradeUsername: null,
```
Comment (richvdh) [inline src/components/structures/MatrixChat.js]: could do with comments for these.

File: src/components/structures/MatrixChat.js [hunk 2]
```diff
@@ -105,6 +105,9 @@ module.exports = React.createClass({
...
+            guestAccessToken: null,
         };
```
Comment (richvdh) [inline src/components/structures/MatrixChat.js]: should we add `register_*` too?

File: src/components/structures/MatrixChat.js [hunk 3]
```diff
@@ -261,13 +264,20 @@ module.exports = React.createClass({
                     newState.register_is_url = payload.params.is_url;
                     newState.register_id_sid = payload.params.sid;
                 }
-                this.replaceState(newState);
+                this.setState(newState);
```
Comment (richvdh) [inline src/components/structures/MatrixChat.js]: do we need to reset `currentRoomAlias` and friends here?

File: src/components/structures/MatrixChat.js (hunk 1 above)
Comment (dbkr) [inline src/components/structures/MatrixChat.js]: done

File: src/components/structures/MatrixChat.js (hunk 2 above)
Comment (dbkr) [inline src/components/structures/MatrixChat.js]: done

File: src/components/structures/MatrixChat.js [hunk 4]
```diff
@@ -233,6 +245,21 @@ module.exports = React.createClass({
...
+            currentRoomAlias: null,
+            currentRoomId: null,
+            viewUserId: null,
+            logged_in: false,
+            ready: false,
+            upgradeUsername: null,
+            guestAccessToken: null,
+       };
+       newState.update(state);
+       this.setState(state);
```
Comment (richvdh) [inline src/components/structures/MatrixChat.js]: s/state/newState/

**Pull request merged**